chips/lowrisc/src/flash_ctrl.rs

```diff
@@ -501,19 +505,21 @@ impl hil::flash::Flash for FlashCtrl<'_> {
             self.configure_info_partition(FlashBank::BANK1, self.region_num);
         }
 
+        self.registers.control.write(
+            CONTROL::OP::PROG
+                + CONTROL::PARTITION_SEL::DATA
+                + CONTROL::NUM.val(((PAGE_SIZE / 4) - 1) as u32)
```

Comment:
The constant `PAGE_SIZE` on L191 is incorrect.  The OpenTitan flash controller IP block is meant to be an abstraction over the physical characteristics of the manufacturer-supplied flash macro.  In the earlgrey implmentation, the [flash page size is 2KiB](https://docs.opentitan.org/hw/ip/flash_ctrl/doc/#flash-physical-controller-features) (256 words of 64 bits each).

The page size is the erase granularity.

The flash programming granularity is the 64-bit word.  The register spec for the flash controller talks about "bus words", which refer to 32-bit words.  The flash programming FIFO is 16 words (e.g. 32-bit words), meaning a single programming operation can program _up to_ 64 bytes into the flash in a single programming operation.